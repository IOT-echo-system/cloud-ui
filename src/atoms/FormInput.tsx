import type {TextFieldProps} from '@mui/material'
import {TextField} from '@mui/material'

export type FormInputType = TextFieldProps

export const FormInput: React.FC<FormInputType> = props => {
  return <TextField variant={props.variant ?? 'outlined'} {...props} />
}
