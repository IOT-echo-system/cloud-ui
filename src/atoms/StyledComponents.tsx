import {Button as MUIButton, Stack, styled} from '@mui/material'
import NextLink from 'next/link'

export const Button = styled(MUIButton)({textTransform: 'none'})

export const FormContainer = styled('form')(({theme}) => ({
  background: theme.palette.background.paper,
  padding: theme.spacing(1),
  boxShadow: theme.shadows[1],
  borderRadius: theme.spacing(1),
  margin: theme.spacing(1),
  justifyContent: 'center',
  width: `calc(100vw - ${theme.spacing(4)})`,
  '&>*': {
    margin: theme.spacing(1.5)
  },
  '&.light': {
    border: `1px solid ${theme.palette.grey[300]}`
  },
  '&.dark': {
    color: theme.palette.common.white,
    border: `1px solid ${theme.palette.grey[700]}`
  },
  [theme.breakpoints.up('sm')]: {
    width: theme.spacing(60),
    padding: theme.spacing(4)
  }
}))

export const CenteredContainer = styled(Stack)(() => ({
  margin: 'auto'
}))

export const Link = styled(NextLink)(() => ({
  textDecoration: 'none',
  '&:hover': {
    textDecoration: 'underline'
  }
}))
