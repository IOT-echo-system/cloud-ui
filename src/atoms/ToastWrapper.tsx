import type {PropsWithChildren} from 'react'
import React, {createRef} from 'react'
import type {SnackbarKey} from 'notistack'
import {SnackbarProvider} from 'notistack'
import {IconButton} from '@mui/material'
import {Close} from '@mui/icons-material'
import {styled} from '@mui/material/styles'

const Toast = styled(SnackbarProvider)(({theme}) => ({
  width: theme.spacing(48)
}))

export const ToastWrapper: React.FC<PropsWithChildren> = ({children}) => {
  const toastRef = createRef<SnackbarProvider>()
  const onClickDismiss = (key: SnackbarKey) => () => {
    toastRef.current?.closeSnackbar(key)
  }

  return (
    <Toast
      maxSnack={5}
      anchorOrigin={{horizontal: 'right', vertical: 'top'}}
      autoHideDuration={5000}
      ref={toastRef}
      aria-multiline={true}
      action={key => (
        <IconButton onClick={onClickDismiss(key)}>
          <Close />
        </IconButton>
      )}
    >
      {children}
    </Toast>
  )
}
