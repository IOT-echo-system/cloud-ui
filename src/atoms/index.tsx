export * from './StyledComponents'
export * from './FormInput'
export * from './ToastWrapper'
export * from './Loader'
