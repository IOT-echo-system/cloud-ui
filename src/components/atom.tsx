import {styled, Button as MUIButton} from '@mui/material'

export const Button = styled(MUIButton)({textTransform: 'none'})
