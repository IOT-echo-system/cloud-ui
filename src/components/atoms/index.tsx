export * from './StyledComponents'
export * from './FormInput'
export * from './ToastWrapper'
export * from './Loader'
export * from './CollapsibleAccordions'
