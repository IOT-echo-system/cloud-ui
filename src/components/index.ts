export {default as Layout} from './Layout'
export * from './Layout'

export {default as SiteWrapper} from './SiteWrapper'
export * from './SiteWrapper'

export * from './atom'
