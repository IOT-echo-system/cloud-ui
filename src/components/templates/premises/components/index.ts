export * from './BoardsView'
export * from './FeedsView'
export * from './PremisesOverview'
export * from './RoutinesView'
export * from './UsersView'
export * from './Widgets'
export * from './ZonesView'
