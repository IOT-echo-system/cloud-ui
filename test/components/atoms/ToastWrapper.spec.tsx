import {render} from '@testing-library/react'
import React from 'react'
import {ToastWrapper} from '../../../src/components/atoms'

describe('Toast Wrapper component test', () => {
  it('should match with the snapshot', () => {
    const TestComponent = () => {
      return <div></div>
    }

    const {container} = render(
      <ToastWrapper>
        <TestComponent />
      </ToastWrapper>
    )

    expect(container).toMatchSnapshot()
  })
})
