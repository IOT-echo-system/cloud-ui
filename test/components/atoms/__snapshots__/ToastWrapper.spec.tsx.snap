// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast Wrapper component test should match with the snapshot 1`] = `
<div>
  <div />
</div>
`;
